src/main/groovy/com/palantir/gradle/shadowjar/ShadowJarConfigurationTask.java

```diff
@@ -88,21 +111,81 @@ public final void run() {
                 })
                 .collect(Collectors.toSet());
 
-        JarFilesRelocator relocator = new JarFilesRelocator(pathsInJars, prefix.get() + ".");
-        shadowJar.relocate(relocator);
+        // The Relocator is responsible for fixing the bytecode at callsites *and* filenames of .class files,
+        // so we have to account for things _calling_ these weird multi-release classes.
+        Set<String> multiReleaseStuff = pathsInJars.stream()
+                .flatMap(input -> splitMultiReleasePath(input).stream().skip(1))
+                .collect(Collectors.toSet());
+
+        Set<String> relocatable = Stream.concat(pathsInJars.stream(), multiReleaseStuff.stream())
+                .filter(path -> !path.equals("META-INF/MANIFEST.MF")) // don't relocate this!
+                .collect(Collectors.toSet());
+
+        shadowJarTask.relocate(new JarFilesRelocator(relocatable, prefix.get() + "."));
+
+        if (!multiReleaseStuff.isEmpty()) {
```

Comment:
Rather than matching classes in what appear to be multi-release locations, could we collect manifests and opt into multi-release if any of our sub-jars do?

Reply:
So i had a crack at that approach, but wasn't a massive fan of their Transformer interface because I don't think it'll compose well if we want to combine other properties (e.g. pdeps) into the same manifest file. 

```java
interface Transformer {

    boolean canTransformResource(FileTreeElement element)

    void transform(TransformerContext context)

    boolean hasTransformedResource()

    void modifyOutputStream(ZipOutputStream jos, boolean preserveFileTimestamps)
}
```

Kinda hoping that we can just do the simplest thing that could possibly work here and then extend/upgrade when we have the need to?

Reply:
Sounds fine -- I was thinking about the edge case where a jar has mr resources that aren't activated via the manifest, however I don't think we can handle that in cases there's a second mr jar that does have the manifest value, so it's not worth thinking about.